### PR TITLE
Instrument `mysqli_commit`

### PR DESCRIPTION
**Pull request by zsistla**
Instrumented `mysqli_commit` to show up in datastore segments.

Handles both:
```
mysqli::commit
mysqli_commit
```
Added test cases to check and manually verified proper behavior in NR1.
Bonus! added parsing of query `CREATE` and `DROP` commands so they don’t show up as `OTHER`.

Fixes issue #95

### CODE REVIEW

**Comment by zsistla** [comment]
image

**Comment by zsistla** [comment]
ok jenkins

**Comment by zsistla** [comment]
ok jenkins

**Comment by zsistla** [comment]
ok jenkins

File: agent/php_internal_instrument.c [hunk 1]
```diff
@@ -734,6 +734,67 @@ NR_INNER_WRAPPER(mysqli_options) {
...
+     * Set the stop time now so that we don't include the explain plan time.
+     */
+    segment->stop_time = nr_txn_now_rel(NRPRG(txn));
+    instance = nr_php_mysqli_retrieve_datastore_instance(mysqli_obj TSRMLS_CC);
+
+    nr_php_txn_end_segment_sql(&segment, sqlstr, sqlstrlen, plan,
+                               NR_DATASTORE_MYSQL, instance TSRMLS_CC);
+
+    nr_explain_plan_destroy(&plan);
+    nr_free(sqlstr);
```
Comment (joshuabenuck) [inline agent/php_internal_instrument.c:789]: Should this free be outside of the if statement so we free `sqlstr` if `segment` is `NULL`?

Comment (zsistla) [reply agent/php_internal_instrument.c:789]: Good call.  I'll put it out of the `if` statement to ensure it is always freed.

**Comment by zsistla** [comment]
ok jenkins

**Comment by zsistla** [comment]
ok jenkins

**Review by joshuabenuck — APPROVED** [review]

**Review by Fahmy-Mohammed — APPROVED** [review]

**Comment by blasto333** [comment]
Just to let you guys know this actually breaks mysqli_commit in production where transactions aren't always commited

**Comment by zsistla** [comment]
Hi @blasto333 , thanks for letting us know.  We're working quickly to get this addressed, and tracking it in #144.
There's a PR up #148 .

**Pull request merged**